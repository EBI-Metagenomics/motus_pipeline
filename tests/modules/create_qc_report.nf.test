nextflow_process {

    name "Test module QC report"
    script "modules/create_qc_report.nf"
    process "QC_REPORT"

    test("Report QC (SE)") {

        when {
            params {
                jsons = "${baseDir}/tests/modules/fixtures/fastp_reports/"
                mode = "single"
            }
            process {
                """
                input[0] = params.mode
                input[1] = channel.fromFilePairs("${params.jsons}/mgtest_fastp.qc.json", checkIfExists: true).map { it[1] }
                input[2] = channel.fromPath("${params.jsons}/dec_report.txt", checkIfExists: true)
                input[3] = channel.fromPath("NO_FILE")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
    test("Report QC and overlap (PE)") {

        when {
            params {
                jsons = "${baseDir}/tests/modules/fixtures/fastp_reports/"
                mode = "paired"
            }
            process {
                """
                input[0] = params.mode
                input[1] = channel.fromPath("${params.jsons}/mgtest_fastp.qc.json", checkIfExists: true)
                input[2] = channel.fromPath("${params.jsons}/dec_report.txt", checkIfExists: true)
                input[3] = channel.fromPath("${params.jsons}/seqprep_output.txt", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
