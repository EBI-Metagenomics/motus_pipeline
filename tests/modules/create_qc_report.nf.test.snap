{
    "Report QC and overlap (PE)": {
        "content": [
            {
                "qc_report": [
                    "qc_summary:md5,d48c8112ad51acb074a65bf1b16c3c07"
                ]
            }
        ],
        "timestamp": "2023-03-30T11:21:33+0000"
    },
    "Report QC (SE)": {
        "content": [
            {
                "qc_report": [
                    "qc_summary:md5,dcbd7acdfe6144da0fcc96ec547414c1"
                ]
            }
        ],
        "timestamp": "2023-03-30T11:21:33+0000"
    }
}